game-core/src/main/java/games/strategy/triplea/delegate/battle/end/NoMoreUnits.java

```diff
@@ -0,0 +1,77 @@
+package games.strategy.triplea.delegate.battle.end;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+@Builder
+public class NoMoreUnits {
```

Comment:
This class could potentially benefit from a Javadoc. Is this a computation class to determine when there is no more units who won? It takes some reverse engineering to see what this does and why it would be used. A good javadoc would helpful give an overview of purpose and functionality to then guide a reader to more quickly understand the implementation. Please also consider adding javadocs to any other classes.